lib/commandbase.rb

```diff
@@ -69,27 +70,44 @@ def load_local_settings
     #
     def tagname_to_ids(array)
       database = Database.instance
+      tag_index = create_tag_index(database)
+      all_ids = database.ids
+      expanded_array = array.map { |arg|
+        if arg =~ /\A\d+\z/
+          # 優先度はID＞タグのため、数字のみ指定されたら
+          # そのIDが存在した場合はIDとみなす
+          id = arg.to_i
+          next id if database[id]
+        end
+        ids =
+          case arg
+          when /\Atag:(.+)\z/
+            # tag:タグ名 は直接タグと指定できる形式
+            # (数字タグとIDがかぶった場合にタグを指定出来るようにするもの)
+            arg = $1
+            tag_index[arg]
+          when /\A\^tag:(.+)\z/
+            # ^tag:タグ名 は除外タグ指定
+            arg = $1
+            indexies = tag_index[arg]
+            indexies.empty? ? [] : all_ids - tag_index[arg]
+          else
+            tag_index[arg]
+          end
+        ids.empty? ? arg : ids
+      }.flatten.uniq
+      array.replace(expanded_array)
+    end
+
+    private def create_tag_index(database)
```

Comment:
__[reek]__
* [UtilityFunction] doesn't depend on instance state (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Utility-Function.md)